Azure/sonic-utilities
# [pfc] Add command line to enable/disable/show PFC

Signed-off-by: Wei Bai baiwei0427@gmail.com

**- What I did**
Current PFC cli only allows operators to enable/disable per-port asymmetric PFC. I extend add new commands to 1) enable/disable PFC on a given priority of a given port and 2) check PFC configuration.  I also link all these commands to <code>show</code> and <code>config</code> utilities.

**- How I did it**
For a interface <code>"EthernetX"</code>, I read the field <code>"pfc_enable"</code>of entry <code>“PORT_QOS_MAP|EthernetX"</code> from config DB table to get its lossless priorities. To enable or disable a specific priority, we just need to modify the field <code>"pfc_enable"</code>.

**- How to verify it**
Enable or disable a priority of a interface. Then check the corresponding entry in config DB. 

**- Previous command output (if the output of a command-line utility has changed)**

**- New command output (if the output of a command-line utility has changed)**
 Show PFC configuration of interface Ethernet0
<pre>
$ pfc show priority Ethernet0


Interface    Lossless priorities
-----------  ---------------------
Ethernet0    3,4
</pre>

<pre>
$ show pfc priority Ethernet0


Interface    Lossless priorities
-----------  ---------------------
Ethernet0    3,4
</pre>
Disable PFC on priority 3 of interface Ethernet0
<pre>
$ pfc config priority off Ethernet0 3


Interface      Lossless priorities
-----------  ---------------------
Ethernet0                        4
</pre>
<pre>
$ config interface pfc priority Ethernet0 3 off


Interface      Lossless priorities
-----------  ---------------------
Ethernet0                        4
</pre>

Disable PFC on priority 8 (invalid value) of interface Ethernet0
<pre>
$ pfc config priority off Ethernet0 8
Usage: pfc config priority [OPTIONS] STATUS INTERFACE PRIORITY

Error: Invalid value for "priority": invalid choice: 8. (choose from 0, 1, 2, 3, 4, 5, 6, 7)
</pre>
<pre>
$ config interface pfc priority Ethernet0 8 off
Usage: config interface pfc priority [OPTIONS] <interface_name> PRIORITY STATUS

Error: Invalid value for "priority": invalid choice: 8. (choose from 0, 1, 2, 3, 4, 5, 6, 7)
</pre>

Enable PFC on priority 3 of interface Ethernet0
<pre>
$ pfc config priority on Ethernet0 3


Interface    Lossless priorities
-----------  ---------------------
Ethernet0    3,4
</pre>
<pre>
$ config interface pfc priority Ethernet0 3 on

Interface    Lossless priorities
-----------  ---------------------
Ethernet0    3,4
</pre>
